app/app.go

```diff
@@ -1781,3 +1788,29 @@ func (d *Deployments) search(
 		return d.inventoryClient.Search(ctx, tid, parms)
 	}
 }
+
+func (d *Deployments) UpdateDeploymentsWithArtifactName(
+	ctx context.Context,
+	artifactName string,
+) error {
+	// first check if there are pending deployments with given artifact name
+	exists, err := d.db.ExistUnfinishedByArtifactName(ctx, artifactName)
+	if err != nil {
+		return errors.Wrap(err, "looking for deployments with given artifact name")
+	}
+	if !exists {
+		return nil
+	}
```

Comment:
I would write it as `else if`; I'm not asking you to change it, just sharing. :)